python/experimental/eminem_importer.py

```
@@ -109,8 +109,8 @@ def load_pickled_data(input_file, verbose=False):
     try:
         # Try gzip first (files may be gzipped regardless of extension)
         try:
-            with gzip.open(input_file, "rb") as f:
-                data = pickle.loads(f.read())
+            with gzip.GzipFile(input_file, "rb") as f:
+                data = pickle.load(f)
```

Ruff (0.15.21)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)